federation/pkg/federation-controller/replicaset/replicasetcontroller.go

```diff
@@ -567,7 +570,7 @@ func (frsc *ReplicaSetController) reconcileReplicaSet(key string) (reconciliatio
 		lrs.Spec.Replicas = &specReplicas
 
 		if !exists {
-			if replicas > 0 {
+			if replicas > 0 && !isPuppet(frs) {
```

Comment:
I did some refactoring, however i would rather keep all the logic in the single place, no separate functions for puppet reconcilation etc.